lib/public/Authentication/LoginCredentials/ICredentials.php

```diff
@@ -53,7 +53,7 @@ public function getLoginName();
 	 *
 	 * @since 12
 	 *
-	 * @return string
+	 * @return string|null
 	 * @throws PasswordUnavailableException
 	 */
 	public function getPassword();
```

Comment:
```suggestion
	public function getPassword() ?string;
```

maybe a good way forward?

Reply:
Yeah but won't that force all implementations to have the same method signature? Public API and all....

Reply:
The only known implementation is in OC\Authentication\LoginCredentials\Credentials

But yeah. Maybe something to do at some point later and then just bump the full API to strict.